### PR TITLE
Update cleanup

### LINKED ISSUE #2989: Cleanup includes in update.cpp
We are including a lot of header files is update.cpp, as we have refactored a lot of these are probably not needed anymore. We should review them and delete what is not needed: https://github.com/equinor/ert/blob/1a45f9e0d774b41647a9417316ab23ff638491bc/libres/lib/analysis/update.cpp#L7

### PR DESCRIPTION
**Pull request by ManInFez**
**Issue**
Resolves #2989 

Cleanup usage of vector_type in update.cpp

remove unused includes in update.cpp

Removing ANALYSIS_UPDATE wf job and related C code

## Pre review checklist

- [ ] Added appropriate labels

Adding labels helps the maintainers when writing release notes, see sections and the
corresponding labels here: https://github.com/equinor/ert/blob/main/.github/release.yml

### CODE REVIEW

File: libres/lib/include/ert/analysis/update.hpp [hunk 1]
```diff
@@ -1,14 +1,50 @@
...
+public:
+    update_data_type() = default;
+    update_data_type(
+        matrix_type *S_in, matrix_type *E_in, matrix_type *D_in,
+        matrix_type *R_in, std::optional<Eigen::MatrixXd> A_in,
+        std::vector<std::pair<matrix_type *, std::shared_ptr<RowScaling>>>
+            A_with_rowscaling_in,
+        const std::vector<bool> &obs_mask_in)
+        : obs_mask(obs_mask_in) {
+        S = *S_in;
```
Comment (joakim-hove) [inline libres/lib/include/ert/analysis/update.hpp:28]: This should take value arguments and move from - i.e. 
```C++
update_date_type(matrix_type S_in, ....) 
: S(std::move(S_in))
{}
```
and the matrix created at callsite must be freed - as it is now there is a quite significant leak (the move will of course reduce the leak a lot).

**Comment by ManInFez** [comment]
test required please!

**Comment by ManInFez** [comment]
Test ert please!

**Comment by ManInFez** [comment]
Test required please!

**Comment by ManInFez** [comment]
Test ert please!

**Comment by oyvindeide** [comment]
Would be great to split @ManInFez
[Remove usage of vector_type in update.cpp](https://github.com/equinor/ert/pull/3005/commits/adafd26c28bd14b5530cdbea151588ace7c99dbb) into two commits, one for deleting, and one for replacing type, other than that this looks good :+1:

**Review by oyvindeide — APPROVED** [review]
LGTM!

**Comment by ManInFez** [comment]
Test required please!

**Comment by ManInFez** [comment]
Test required please!

**Comment by ManInFez** [comment]
test ert please!

**Pull request merged**